### PR TITLE
feat: implemented Overview UI

### PR DESCRIPTION
**Pull request by Teja2045**
#879

### CODE REVIEW

File: frontend/src/app/(routes)/(overview)/overview-components/PageAd.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import Image from "next/image";
+import React from "react";
+
+const PageAd = () => {
+  return (
+    <div className="my-10 cursor-pointer brightness-50">
+      <Image className="w-full" src="/page-ad-sample.png" width={1000} height={80} alt="Ad" />
```
Comment (aleem1314) [inline frontend/src/app/(routes)/(overview)/overview-components/PageAd.tsx:7]:
```suggestion
      <Image className="w-full" src="https://dummyimage.com/100X80/000/fff&text=Overview+Ad1" width={1000} height={80} alt="Ad" />
```

You can use this URL to render images with arbitrary width and hight.

File: frontend/src/app/(routes)/(overview)/overview-components/SideAd.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import Image from "next/image";
+import React from "react";
+
+const SideAd = () => {
+  return (
+    <div className="my-10 cursor-pointer text-white">
+      <Image src="/side-ad-sample.png" height={216} width={326} alt="Ad" />
```
Comment (aleem1314) [inline frontend/src/app/(routes)/(overview)/overview-components/SideAd.tsx:7]: ditto

File: frontend/src/app/(routes)/(overview)/overview-components/WalletSummery.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import { displayAmountInDollars } from '@/utils/util';
+import Image from 'next/image';
+import React from 'react';
+type AssetSummary = { icon: string; alt: string; type: string; amount: string };
+
+const WalletSummery = (props: {
+  balanceAmount: number;
+  stakedAmount: number;
+  rewardsAmount: number;
```
Comment (aleem1314) [inline frontend/src/app/(routes)/(overview)/overview-components/WalletSummery.tsx:9]:
```suggestion
  available: number;
  staked: number;
  rewards: number;
```
Use short names

File: frontend/src/utils/util.ts [hunk 4]
```diff
@@ -68,3 +67,24 @@ export const getSelectedPartFromURL = (urlParts: string[]): string => {
       return 'Overview';
   }
 };
+
+export const displayAmountInDollars = (amount: number): string => {
+  if (amount === 0) return '$ 0';
+  if (amount < 0.001) return '< $ 0.001';
+  return '$ ' + amount.toFixed(3);
+};
```
Comment (aleem1314) [inline frontend/src/utils/util.ts:75]:
```suggestion
export const formatDollarAmount = (amount: number): string => {
  if (amount === 0) return '$0';
  if (amount < 0.1) return '< $0.1';
  return '$ ' + amount.toFixed(2);
};
```

File: frontend/src/utils/util.ts [hunk 5]
```diff
@@ -68,3 +67,24 @@ export const getSelectedPartFromURL = (urlParts: string[]): string => {
...
+
+export const displayAmount = (amount: number): string => {
+  return amount === 0 ? '0' : amount.toFixed(3);
+};
```
Comment (aleem1314) [inline frontend/src/utils/util.ts:79]:
```suggestion
export const formatAmount = (amount: number): string => {
  return amount === 0 ? '0' : amount.toFixed(2);
};
```

File: frontend/src/utils/util.ts [hunk 6]
```diff
@@ -68,3 +67,24 @@ export const getSelectedPartFromURL = (urlParts: string[]): string => {
...
+export const displayAmountWithDenom = (
+  amount: number,
+  denom: string
+): string => {
+  let parsedAmount;
+  if (amount === 0) parsedAmount = '0';
+  else if (amount < 0.001) parsedAmount = '< 0.001';
+  else parsedAmount = amount.toFixed(3);
+  return parsedAmount + ' ' + denom;
+};
```
Comment (aleem1314) [inline frontend/src/utils/util.ts:90]:
```suggestion
export const formatCoin = (
  amount: number,
  denom: string
): string => {
  let parsedAmount;
  if (amount === 0) parsedAmount = '0';
  else if (amount < 1) parsedAmount = '< 1';
  else parsedAmount = amount.toFixed(2);
  return parsedAmount + ' ' + denom;
};
```

**Comment by aleem1314** [comment]
```
472-8cd38b62f3a6056b.js:1 Error: Given amount is not a safe integer. Consider using a string instead to overcome the limitations of JS numbers.
    at ed (174-982bb66ea6918727.js:1:13215)
    at B (https://resolute-hs38c3b47-my-apps-projects.vercel.app/_next/static/chunks/app/(routes)/(/overview)/page-8c17372173653ded.js:1:114107)
    at https://resolute-hs38c3b47-my-apps-projects.vercel.app/_next/static/chunks/app/(routes)/(/overview)/page-8c17372173653ded.js:1:115521
    at Array.forEach (<anonymous>)
    at https://resolute-hs38c3b47-my-apps-projects.vercel.app/_next/static/chunks/app/(routes)/(/overview)/page-8c17372173653ded.js:1:115441
    at Object.r8 [as useMemo] (fd9d1056-8548210fd6e97332.js:1:46675)
    at t.useMemo (472-8cd38b62f3a6056b.js:1:97140)
    at k (https://resolute-hs38c3b47-my-apps-projects.vercel.app/_next/static/chunks/app/(routes)/(/overview)/page-8c17372173653ded.js:1:115414)
    at I (https://resolute-hs38c3b47-my-apps-projects.vercel.app/_next/static/chunks/app/(routes)/(/overview)/page-8c17372173653ded.js:1:120520)
    at rb (fd9d1056-8548210fd6e97332.js:1:40330)
```

Can you take a look into this issue? Page is breaking with above error.

**Comment by Teja2045** [comment]
> ```
> 472-8cd38b62f3a6056b.js:1 Error: Given amount is not a safe integer. Consider using a string instead to overcome the limitations of JS numbers.
>     at ed (174-982bb66ea6918727.js:1:13215)
>     at B (https://resolute-hs38c3b47-my-apps-projects.vercel.app/_next/static/chunks/app/(routes)/(/overview)/page-8c17372173653ded.js:1:114107)
>     at https://resolute-hs38c3b47-my-apps-projects.vercel.app/_next/static/chunks/app/(routes)/(/overview)/page-8c17372173653ded.js:1:115521
>     at Array.forEach (<anonymous>)
>     at https://resolute-hs38c3b47-my-apps-projects.vercel.app/_next/static/chunks/app/(routes)/(/overview)/page-8c17372173653ded.js:1:115441
>     at Object.r8 [as useMemo] (fd9d1056-8548210fd6e97332.js:1:46675)
>     at t.useMemo (472-8cd38b62f3a6056b.js:1:97140)
>     at k (https://resolute-hs38c3b47-my-apps-projects.vercel.app/_next/static/chunks/app/(routes)/(/overview)/page-8c17372173653ded.js:1:115414)
>     at I (https://resolute-hs38c3b47-my-apps-projects.vercel.app/_next/static/chunks/app/(routes)/(/overview)/page-8c17372173653ded.js:1:120520)
>     at rb (fd9d1056-8548210fd6e97332.js:1:40330)
> ```
> 
> Can you take a look into this issue? Page is breaking with above error.

I will look into it, how to reproduce?

**Review by aleem1314 — APPROVED** [review]

**Pull request merged**